lora-tutorials/lora-notecard-application/lora-notecard-application.ino

```diff
@@ -0,0 +1,244 @@
+
+    J *req = notecard.newRequest("env.get");
+    JAddNumberToObject(req, "time", environmentModifiedTime);
+    J *rsp = notecard.requestAndResponse(req);
+    if (rsp != NULL) {
+      if (!notecard.responseError(rsp)) {
+        refreshEnvironmentVarCache(rsp);
+      }
+      notecard.deleteResponse(rsp);
+    }
+  }
+
+  // On a periodic basis, check the inbound queue for messages.  In a
+  // real-world application, this would be checked using a frequency
+  // commensurate with the required inbound responsiveness. For the most
+  // common "periodic" mode applications, this might be daily or weekly.  In
+  // this example, where we are using "continuous" mode, we check quite often
+  // for demonstratio purposes.
+  static unsigned long nextInboundPollMs = 0;
+  if (millis() > nextInboundPollMs)
+  {
+      nextInboundPollMs = millis() + (INBOUND_NOTEFILE_POLL_SECS * 1000);
+
+      // Process all pending inbound requests
+      while (true)
+      {
+          // Get the next available note from our inbound queue notefile,
+          // deleting it
+          J *req = notecard.newRequest("note.get");
+          JAddStringToObject(req, "file", INBOUND_NOTEFILE);
+          JAddBoolToObject(req, "delete", true);
+          J *rsp = notecard.requestAndResponse(req);
+          if (rsp != NULL)
+          {
+              // If an error is returned, this means that no response is
+              // pending.  Note that it's expected that this might return
+              // either a "note does not exist" error if there are no pending
+              // inbound notes, or a "file does not exist" error if the
+              // inbound queue hasn't yet been created on the service.
+              if (notecard.responseError(rsp))
+              {
+                  notecard.deleteResponse(rsp);
+                  break;
+              }
+
+              // Get the note's body
+              J *body = JGetObject(rsp, "body");
+              if (body != NULL)
+              {
+
+                  // Simulate Processing the response here
+                  notecard.logDebug("INBOUND REQUEST: ");
+                  notecard.logDebug(JGetString(body, TVAR_POLL));
+                  notecard.logDebug("\n\n");
+              }
+          }
+          notecard.deleteResponse(rsp);
+      }
+  }
+}
+
+// Re-load all env vars, returning the modified time
+void refreshEnvironmentVarCache(J *rsp)
+{
+  // Update the env modified time
+  environmentModifiedTime = JGetNumber(rsp, "time");
+
+  // Update the environment
+  J *body = JGetObject(rsp, "body");
+  if (body == NULL) {
+    return;   
+  }
+
+  // Update heartbeat period
+  envHeartbeatMins = JAtoN(JGetString(body, TVAR_MINUTES), NULL);
+  if (envHeartbeatMins == 0) {
+    envHeartbeatMins = DATA_SEND_MINS;
+  }
+
+  // See if the notecard has been rebooted, and reboot ourselves if so
+  rsp = notecard.requestAndResponse(notecard.newRequest("card.status"));
```

Comment:
This is a dangerous habit.

How about using a new variable name, instead of reusing `rsp`?